### DISCUSSION TITLE
Issues writing back to certain tags.

### DISCUSSION THREAD

**Discussion by cmrncrick** [Hugovdberg/PIconnect#776] [Q&A]
I am having a problem writing back to certain tags and I'm not sure what the issue is. These tags are Int32. The following is my code that I'm using to write to the tag:

```python
try:

    point = server.search(pi_tag_name)[0]
    print(f"** POINT ** {point}")
    point.update_value(
        value,
        write_time,
        UpdateMode.NO_REPLACE,
        BufferMode.BUFFER_IF_POSSIBLE)

    logger.info(
        f"{pi_tag_name} successfully written with value of {value}.")

except Exception as e:

    logger.critical(
        f"Exception writing back to Pi for {pi_tag_name}.")
    logger.critical(f"{e}\n")
```

value is a value that I have casted into an integer. write_time is datetime.now().

The exception that I get back in my logger is:

```
2024-07-08 09:46:06,920: CRITICAL -- 204 -- Exception writing back to Pi for SMACT.SWARF_BB_DLY.NUM.
2024-07-08 09:46:06,922: CRITICAL -- 206 -- Cannot convert value to a PI equivalent type for writing.

   at OSIsoft.AF.PI.PIPoint.CoerceAFValueThenAddToMDAEvents(AFValue afVal, EventWriteMode updateMode, Events events)

   at OSIsoft.AF.PI.PIPoint.ConvertAFValueToMDAEvents(AFValue afval, AFUpdateOption option)

   at OSIsoft.AF.PI.PIPoint.UpdateValue(AFValue value, AFUpdateOption option, AFBufferOption bufferOption)
```
Can anyone help me understand why this is happening please?

**Comment by Hugovdberg** [comment]
.NET is a lot stricter when it comes to data types, could it be that you try to write a float to the integer point? Does the error still occur if you explicitly convert your value to an `int` before writing?

**Reply by cmrncrick** [reply]
> .NET is a lot stricter when it comes to data types, could it be that you try to write a float to the integer point? Does the error still occur if you explicitly convert your value to an `int` before writing?

I couldn't get anything to write to those specific tags. I even made sure to convert the value to an int before trying to write. It was really weird and I honestly don't understand why I couldn't write to it. 

What I ended up doing was changing the tag configuration itself from int32 to float64 and after that I was able to write to it. Not sure why that is??
